packages/devtools_app/lib/src/screens/deep_link_validation/deep_link_list_view.dart

```diff
@@ -36,67 +36,59 @@ class _DeepLinkListViewState extends State<DeepLinkListView>
       controller.selectedProject.value!.androidVariants;
 
   @override
-  void initState() {
-    super.initState();
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    initController();
     callWhenControllerReady((_) {
       int releaseVariantIndex = controller
           .selectedProject.value!.androidVariants
           .indexWhere((variant) => variant.toLowerCase().contains('release'));
       // If not found, default to 0.
       releaseVariantIndex = max(releaseVariantIndex, 0);
-      controller.selectedVariantIndex = releaseVariantIndex;
+      controller.selectedVariantIndex.value = releaseVariantIndex;
     });
   }
 
   @override
-  void didChangeDependencies() {
-    super.didChangeDependencies();
-    if (!initController()) return;
-  }
-
-  Widget _buildTopPanel() {
-    return Row(
-      mainAxisAlignment: MainAxisAlignment.end,
-      children: [
-        Padding(
-          padding: const EdgeInsets.all(defaultSpacing),
-          child: _buildAndroidVariantDropDown(),
-        ),
-      ],
+  Widget build(BuildContext context) {
+    return DefaultTabController(
+      length: TableViewType.values.length,
+      child: const Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          _DeepLinkListViewTopPanel(),
+          SizedBox(height: denseSpacing),
+          Expanded(child: _DeepLinkListViewMainPanel()),
+        ],
+      ),
     );
   }
+}
 
-  Widget _buildAndroidVariantDropDown() {
-    final List<DropdownMenuEntry<int>> colorEntries =
-        <DropdownMenuEntry<int>>[];
-    for (int i = 0; i < androidVariants.length; i += 1) {
-      colorEntries.add(
-        DropdownMenuEntry<int>(
-          value: i,
-          label: androidVariants[i],
-        ),
-      );
-    }
-    return DropdownMenu<int>(
-      initialSelection: controller.selectedVariantIndex,
-      label: const Text('Android Variant'),
-      dropdownMenuEntries: colorEntries,
-      onSelected: (int? index) {
-        setState(() {
-          controller.selectedVariantIndex = index!;
-        });
-      },
-    );
+class _DeepLinkListViewMainPanel extends StatefulWidget {
```

Comment:
same comment here about making this stateless since we aren't storing anything in the state but a `controller` object from `ProvidedControllerMixin`.  You can just look up the controller in the build method directly or pass in as a parameter.